### PR TITLE
Add github workflow to update status on Label Action

### PR DESCRIPTION
**Pull request by ScottSuarez**
This change is being made as part of the magician-breaking-change detection work. 

We are planning to block merging on the status check failing but want to allow the developer the ability to override that. This enables that through a switch of a status check after applying a specific label.

companion to:
https://github.com/GoogleCloudPlatform/magic-modules/pull/6642

go/magician-breaking-change

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Update Breaking Change Status
+on:
+  pull_request:
+    types:
+      - labeled
+jobs:
+  add-comment:
+    if: github.event.label.id == '4598495472'
```
Comment (rileykarson) [inline .github/workflows/docker-image.yml:8]: Can you add a comment indicating what this magic number corresponds to?

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      statuses: write
+    steps:
+    - uses: actions/checkout@v2
+    - name: Allow breaking change to merged
+      uses: Sibz/github-status-action@v1
```
Comment (rileykarson) [inline .github/workflows/docker-image.yml:16]: Are there alternatives to taking a dependency on an arbitrary community plugin here?

**Review by rileykarson — COMMENTED** [review]
Handing review off to @melinath. Left a couple comments on the way.

One thing that's not clear in this PR alone is the context behind it. Why do we want to update the PR status based on an arbitrary label number? That's a question that's answered by the external context I've got, but adding a description to the issue and linking to issues / bugs / related PRs can make the changes more clear to members of the team in the future.

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (ScottSuarez) [inline .github/workflows/docker-image.yml:8]:
```suggestion
    if: github.event.label.id == '4598495472' # The unique identifier for the override-breaking change label.. used in place of name so the name can be modified without breaking usage
```

File: .github/workflows/docker-image.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+name: Update Breaking Change Status
+on:
+  pull_request:
+    types:
+      - labeled
+jobs:
+  add-comment:
+    if: github.event.label.id == '4598495472' # The unique identifier for the override-breaking change label.. used in place of name so the name can be modified without breaking usage
```
Comment (ScottSuarez) [inline .github/workflows/docker-image.yml:8]:
```suggestion
    if: github.event.label.id == '4598495472' # The unique identifier for the override-breaking-change label.. used in place of name so the name can be modified without breaking usage
```

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (ScottSuarez) [reply .github/workflows/docker-image.yml]: done!

File: .github/workflows/docker-image.yml (hunk 2 above)
Comment (ScottSuarez) [reply .github/workflows/docker-image.yml:16]: I'll investigate!

Comment (melinath) [reply .github/workflows/docker-image.yml]: go/github-actions and go/github-docs/security-recommendations are relevant here

Comment (melinath) [reply .github/workflows/docker-image.yml]: looks like this was changed

**Comment by ScottSuarez** [comment]
looks to be working ^^

File: .github/workflows/docker-image.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+on:
+  pull_request:
+    types:
+      - labeled
+jobs:
+  update-breaking-change-status:
+    if: github.event.label.id == '4598495472' # The unique identifier for the override-breaking-change label.. used in place of name so the name can be modified without breaking usage
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
```
Comment (melinath) [inline .github/workflows/docker-image.yml:11]: just to double-check, does it need these permissions if it's only updating the `statuses` endpoint?

**Review by melinath — APPROVED** [review]

File: .github/workflows/docker-image.yml (hunk 4 above)
Comment (ScottSuarez) [inline .github/workflows/docker-image.yml:11]:
```suggestion
      pull-requests: read
```

Comment (ScottSuarez) [reply .github/workflows/docker-image.yml]: Should be doable with read only

**Pull request merged**